pkg/leaderelection/leaderelection.go

```diff
@@ -143,20 +145,28 @@ func (le *LeaderElector) Run(ctx context.Context) error {
 				le.logger.Infof("backup-restore changed the state from %v to %v", StateUnknown, le.CurrentState)
 			} else if !isLeader && le.CurrentState == StateFollower {
 				le.logger.Debugf("backup-restore currentState: %v", le.CurrentState)
+
+				// If learner(non-voting member) is present in a cluster(size>1)
+				// then promote the learner to voting member.
+				if isLearner && le.PromoteCallback != nil {
+					le.logger.Info("found a learner(non-voting) member in a cluster...")
+					le.PromoteCallback.Promote(ctx, le.logger)
+				}
 			}
 		}
 	}
 }
 
 // IsLeader checks whether the current instance of backup-restore is leader or not.
-func IsLeader(ctx context.Context, etcdConnectionConfig *brtypes.EtcdConnectionConfig, etcdConnectionTimeout time.Duration, logger *logrus.Entry) (bool, error) {
+// It also returns the boolean indicating the presence of learner(non-voting) member.
+func IsLeader(ctx context.Context, etcdConnectionConfig *brtypes.EtcdConnectionConfig, etcdConnectionTimeout time.Duration, logger *logrus.Entry) (bool, bool, error) {
```

Comment:
Rename function since it no longer checks for only leadership status, but also learner status.